src/main/java/com/karl/ProtocolFactoryable.java

```diff
@@ -0,0 +1,7 @@
+package com.karl;
+
+import com.karl.wrappers.Connectable;
+
+public interface ProtocolFactoryable {
```

Comment:
Nice use of interface here!
Naming interfaces to start with the letter I will make a readability improvement when reading it from other classes. This is a useful convention.